src/aldente.cpp

```diff
@@ -75,12 +77,30 @@ void Aldente::start_game_loop() {
     // Have window fire off a resize event to update all interested systems.
     window.broadcast_size();
 
+    // Game logic.
+    Phase::curr_phase = PhaseType::BUILD;
+
+    NetworkManager::connect();
+
+    events::joystick_event.connect([&](events::JoystickData d) {
+        if (Phase::curr_phase == PhaseType::BUILD) {
+            if (d.is_button == true && d.input == 0 && d.state == 0) {
+                BuildPhase::is_menu = false;
```

Comment:
Instead of having to demux off of the ``curr_phase`` ``PhaseType`` enum, you can make use of ``GameState`` by making it have a ``Phase *curr_phase`` (thus making it *the* GameState (i.e. static)). Thereby using inheritance/class hierarchy of ``Phase`` class to demux, and ``virtual setup`` (where a Phase subscribes to events) and ``update`` (as you have already) functions.